### PR TITLE
More Rust dependency updates for upcoming v1.13 release

### LINKED ISSUE #2763: v1.13.0 `aws-sdk-rust`updates
For the v1.13.0, as we approach a code freeze, let's upgrade the `aws-sdk-rust` dependency to capture any new changes and  fixes

### PR DESCRIPTION
**Pull request by jpculp**
**Issue number:**

Closes #2763

**Description of changes:**

Updates AWS SDK for Rust, `aws-smithy`, `coldsnap`, and `tough`, while addressing any breaking changes.

Also includes changes to remove an unnecessary `time` 0.1.x dependency, although it won't be completely removed until the next release of `testsys`/`bottlerocket-types`.

**Testing done:**

Built and launched `aws-k8s-1.24` (x86_64) and ran through some basic operations.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: tools/deny.toml [hunk 1]
```diff
@@ -58,14 +58,25 @@ multiple-versions = "deny"
...
 
 skip = [
+    # num_cpus uses an old version of hermit-abi
+    { name = "hermit-abi", version = "=0.2.6" },
+    # tungstenite other crates use an old version of base64
+    { name = "base64", version = "=0.13.1" },
+    # governor uses an old version of wasi
+    { name = "wasi", version = "=0.10.2" },
+    # testsys uses an old version of time
+    { name = "time", version = "=0.1.43" },
```
Comment (jpculp) [inline tools/deny.toml:68]: We'll want to remove this whenever we update testsys.

File: tools/Cargo.lock [hunk 2]
```diff
@@ -74,10 +74,11 @@ checksum = "64cb94155d965e3d37ffbbe7cc5b82c3dd79dd33bd48e536f73d2cfb8d85506f"
 
 [[package]]
 name = "assert-json-diff"
-version = "2.0.2"
+version = "1.1.0"
```
Comment (jpculp) [inline tools/Cargo.lock:77]: This is a dev-dependency for `infrasys` and `aws-smithy-protocol-test`. From what I can tell the `infrasys` tests don't mine dropping to the same version as `aws-smithy`, so we can avoid pulling multiple versions.

**Review by stmcginnis — APPROVED** [review]
Looks fine to me.

**Review by webern — APPROVED** [review]
Please test download repo before merging. :+1:

**Review by jpmcb — APPROVED** [review]

**Comment by jpculp** [comment]
> Please test download repo before merging. 👍

I did `validate repo`, but forgot to mention it in the description. Ran it again and got this output:
```
fish ❯ cargo make validate-repo
[cargo-make] INFO - cargo make 0.36.4
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: validate-repo
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Running Task: setup
[cargo-make] INFO - Running Task: fetch-sources
[cargo-make] INFO - Running Task: tuftool
[cargo-make] INFO - Running Task: publish-setup-tools
[cargo-make] INFO - Running Task: publish-setup
22:15:31 [INFO] Found infra config at path: /home/jpculp/workspace/bottlerocket/Infra.toml
[cargo-make] INFO - Running Task: setup
[cargo-make] INFO - Running Task: fetch-sources
[cargo-make] INFO - Running Task: publish-tools
[cargo-make] INFO - Running Task: validate-repo
22:15:57 [INFO] Found infra config at path: /home/jpculp/workspace/bottlerocket/Infra.toml
22:15:57 [INFO] Loaded TUF repo: https://updates.bottlerocket.aws/2020-07-07/aws-k8s-1.24/x86_64
22:15:57 [INFO] Downloading target: migrate_v1.11.0_aws-creds.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_dns-settings.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_aws-control-container-v0-6-4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.1-5d27ae74-boot.ext4.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_reboot-to-reconcile-setting.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_public-admin-container-v0-9-4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.1-104f8e0f-root.verity.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.12.0-6ef1139f-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.0-b530f308-root.verity.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.10.1-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.12.0-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.0-b530f308-boot.ext4.lz4
22:15:57 [INFO] Downloading target: aws-k8s-1.24-x86_64-kmod-kit-v1.10.0.tar.xz
22:15:57 [INFO] Downloading target: migrate_v1.10.1_container-runtime.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_public-control-container-v0-6-3.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.0-3949354b.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.1-5d27ae74-root.verity.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_kubelet-new-config-files.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.1_container-runtime-metadata.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_public-control-container-v0-6-4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.10.0-data.img.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_kubelet-tls-files.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_dns-settings-metadata.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_kubelet-log-level.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.11.1.img.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_ecs-additional-configurations.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.1-5d27ae74-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.10.1.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.1-104f8e0f-data.img.lz4
22:15:57 [INFO] Downloading target: aws-k8s-1.24-x86_64-kmod-kit-v1.12.0.tar.xz
22:15:57 [INFO] Downloading target: migrate_v1.11.0_aws-creds-metadata.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.0-3949354b-boot.ext4.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_public-admin-container-v0-9-3.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_public-control-container-v0-7-0.lz4
22:15:57 [INFO] Downloading target: aws-k8s-1.24-x86_64-kmod-kit-v1.10.1.tar.xz
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.0-b530f308-root.ext4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.0-b530f308-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.1-5d27ae74.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.10.0.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.11.0.img.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_aws-control-container-v0-6-3.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.12.0-6ef1139f-boot.ext4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.1-104f8e0f-root.ext4.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_public-admin-container-v0-9-2.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_kubelet-tls-config.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_oci-defaults-setting-metadata.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.0-3949354b-root.verity.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_aws-admin-container-v0-9-4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.12.0-6ef1139f.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.0-3949354b-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.0-3949354b-root.ext4.lz4
22:15:57 [INFO] Downloading target: aws-k8s-1.24-x86_64-kmod-kit-v1.11.0.tar.xz
22:15:57 [INFO] Downloading target: migrate_v1.12.0_add-k8s-autoscaling-warm-pool-setting-metadata.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_aws-control-container-v0-7-0.lz4
22:15:57 [INFO] Downloading target: migrate_v1.10.0_aws-admin-container-v0-9-2.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.12.0-6ef1139f-root.ext4.lz4
22:15:57 [INFO] Downloading target: aws-k8s-1.24-x86_64-kmod-kit-v1.11.1.tar.xz
22:15:57 [INFO] Downloading target: migrate_v1.11.0_aws-config-settings.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_aws-admin-container-v0-9-3.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.12.0-6ef1139f-root.verity.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.0-b530f308.img.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_k8s-private-pki-path.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_oci-defaults-setting.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.1-104f8e0f.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.12.0.img.lz4
22:15:57 [INFO] Downloading target: migrate_v1.12.0_add-k8s-autoscaling-warm-pool-setting.lz4
22:15:57 [INFO] Downloading target: migrate_v1.11.0_credential-providers.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.11.0-data.img.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.11.1-104f8e0f-boot.ext4.lz4
22:15:57 [INFO] Downloading target: manifest.json
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-1.10.1-5d27ae74-root.ext4.lz4
22:15:57 [INFO] Downloading target: bottlerocket-aws-k8s-1.24-x86_64-v1.11.1-data.img.lz4
[cargo-make] INFO - Build Done in 52.74 seconds.
```

**Review by stmcginnis — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Pull request merged**